src/SDK/MonitoringApplianceEmulator/Controls/HypertextBlock.cs

```diff
@@ -0,0 +1,135 @@
+
+        #region Dependency Properties
+
+        /// <summary>
+        /// Gets or sets the hypertext.
+        /// </summary>
+        public string Hypertext
+        {
+            get
+            {
+                return (string)this.GetValue(HypertextProperty);
+            }
+
+            set
+            {
+                this.SetValue(HypertextProperty, value);
+            }
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Occurs when the <see cref="Hypertext"/> dependency property was changed.
+        /// This method replace all link patterns within the new assigned hypertext  <see cref="Hyperlink"/> elements.
+        /// </summary>
+        /// <param name="dependencyObject">The dependency object</param>
+        /// <param name="eventArgs">The event args</param>
+        private static void OnHypertextPropertyChanged(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs eventArgs)
+        {
+            if (!(dependencyObject is HypertextBlock hypertextTextBlock))
+            {
+                throw new ArgumentException($"The dependency object must be of type {typeof(HypertextBlock)}, but it is from type {dependencyObject.GetType()}.", nameof(dependencyObject));
+            }
+
+            if (!(eventArgs.NewValue is string hypertext))
+            {
+                throw new ArgumentException($"The new value must be of type {typeof(string)}, but it is from type {eventArgs.NewValue.GetType()}.", nameof(eventArgs));
+            }
+
+            Match match = HtmlHyperlinkRegex.Match(hypertext);
+
+            // In case there are no links in the text, just add it
+            if (!match.Success)
+            {
+                hypertextTextBlock.Inlines.Add(hypertext);
+            }
+
+            // Go over all link pattern matches and replace it with a hyperlink element
+            Match previousMatch = null;
+            while (match.Success)
+            {
+                // First, add text before the current match
+                int startIndexOfTextBeforeMatch = previousMatch == null ?
+                    0 :
+                    previousMatch.Index + previousMatch.Length;
+
+                int lengthOfTextBeforeMatch = previousMatch == null ?
+                    match.Index :
+                    match.Index - (previousMatch.Index + previousMatch.Length);
+
+                string textBeforeMatch = hypertext.Substring(startIndexOfTextBeforeMatch, lengthOfTextBeforeMatch);
+
+                hypertextTextBlock.Inlines.Add(textBeforeMatch);
+
+                // Second, convert the match to hyperlink and add it
+                string url = match.Groups["href"].Value.Trim('\'', '\"');
+                string linkDisplayText = match.Groups["linkText"].Value;
+
+                Hyperlink link = new Hyperlink { IsEnabled = true };
```

Comment:
>Hyperlink link = new Hyperlink { IsEnabled = true }; [](start = 16, length = 52)

Can the user copy the URL (with a right click)? If not, could you open a US/Task to enable it?

Reply:
no. Added task 1391316

---
In reply to: [234441516](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/25#discussion_r234441516) [](ancestors = 234441516)